scripts/mkstatic.py

```diff
@@ -839,9 +852,39 @@ def write(self, metadata_request, metadata_define, arguments):
                             elif self.parents[ccpp_stage][parent_standard_name].intent == 'out' and not var.intent == 'out':
                                 self.parents[ccpp_stage][parent_standard_name].intent = 'inout'
 
-                        # Add to argument list
-                        arg = '{local_name}={var_name},'.format(local_name=var.local_name, 
-                                                                var_name=local_vars[var_standard_name]['var_local_name'])
+                        # Add necessary actions before/after while populating the subroutine's argument list
+                        if var.actions['out']:
+                            if local_vars[var_standard_name]['var_local_name'] in tmpvars.keys():
+                                # If the variable already has a local variable (tmpvar), reuse it
+                                tmpvar = tmpvars[local_vars[var_standard_name]['var_local_name']]
+                            else:
+                                # Add a local variable (tmpvar) for this variable
+                                tmpvar_cnt += 1
+                                tmpvar = copy.deepcopy(var)
+                                tmpvar.local_name = 'tmpvar{0}'.format(tmpvar_cnt)
```

Comment:
This sequence (862 - 864) is the same as lines 589 - 591 in mkcap.py. I think it would be nice if there was one place for creating this sort of variable, perhaps a separate method of the `Var` class. Just an idea.
Of course, this whole block looks a lot like the equivalent code in mkcap.py so maybe there is a compelling reason for all the duplication?

Reply:
You are right, this and several other parts of the code-generation block are duplicates. There are however differences that would have made a "unification" of the code slightly complicated. Given that the dynamic build will soon be gone, I didn't see the point in making this effort.

However, I did note in the PR that I would like to separate the Var class into its own Python module, because this would benefit the static build as well, as long at it exists.